apps/platform-app/src/lib/langchain.ts

```diff
@@ -0,0 +1,194 @@
+'use server'
+import { type AIMessage } from '@langchain/core/messages'
+import { ChatGroq } from '@langchain/groq'
+import { END, START, StateGraph } from '@langchain/langgraph'
+import { ToolNode } from '@langchain/langgraph/prebuilt'
+
+import {
+  sendProfileApprovedEmail,
+  sendProfileRejectedEmail,
+} from '@/backend/mailing/mailing.service'
+import { updateProfileById } from '@/backend/profile/profile.service'
+import { getUserById } from '@/backend/user/user.service'
+import { type UserWithProfileAndGH } from '@/backend/user/user.types'
+import { getContextVariable, setContextVariable } from '@langchain/core/context'
+import { type BaseMessage } from '@langchain/core/messages'
+import { tool } from '@langchain/core/tools'
+import { Annotation, type LangGraphRunnableConfig } from '@langchain/langgraph'
+import { z } from 'zod'
+
+const StateAnnotation = Annotation.Root({
+  messages: Annotation<BaseMessage[]>({
+    reducer: (x, y) => x.concat(y),
+  }),
+  user: Annotation<UserWithProfileAndGH>(),
+})
+
+const evaluationModel = new ChatGroq({
+  model: 'llama-3.1-70b-versatile',
+  temperature: 0.2,
+})
+
+const executionModel = new ChatGroq({
+  model: 'llama3-groq-8b-8192-tool-use-preview',
+  temperature: 0.1,
+})
+
+const acceptProfileTool = tool(
+  async (_, config: LangGraphRunnableConfig) => {
+    const userId = config.configurable?.userId
+    const currentState =
+      getContextVariable<typeof StateAnnotation.State>('currentState')
+
+    // updateProfilePublishingState(userId, 'APPROVED')
+
+    if (currentState?.user) {
+      if (currentState?.user.profile) {
+        updateProfileById(currentState.user.profile.id, { state: 'APPROVED' })
+      }
+
+      sendProfileApprovedEmail(
+        currentState?.user.email,
+        currentState?.user.githubDetails?.username || '',
+      )
+    }
+    return `The profile has been accepted. UserId: ${userId} `
+  },
+  {
+    name: 'accept_profile',
+    description: 'Accept user apply and save his profile in our database',
+    schema: z.object({}),
+  },
+)
+
```

Comment:
**suggestion**: Consider adding error handling for the `rejectProfileTool` to manage potential failures in updating the profile or sending emails.